Sources/UIAlertControllerView.swift

```diff
@@ -43,48 +57,103 @@ class UIAlertControllerView: UIView {
+            guard let largestSubviewWidth = subviews.map({ $0.frame.width }).max() else {
+                assertionFailure("no subviews exist to determine largestSubviewWidth")
+                return minWidth
+            }
+            return min(maxWidth, max(largestSubviewWidth, minWidth))
+        }()
     }
 
     override func layoutSubviews() {
-        switch style {
-        case .actionSheet: layoutAsActionSheet()
-        default: assertionFailure(
-            "The UIAlertControllerStyle style, \(style), is not implemented yet!")
+        subviews.forEach { $0.sizeToFit() }
+
+        header.frame.origin = CGPoint(x: horizontalPadding, y: verticalPadding)
+        header.bounds.width = subviewWidth
+        header.sizeToFit()
+        header.textAlignment = .left // sizeToFit() seems to change textAlignment
+
+        text?.frame.origin.x = horizontalPadding
+        text?.frame.width = subviewWidth
+        text?.sizeToFit()
+        text?.frame.minY = header.frame.maxY + verticalPadding
+
+        if style == .alert {
+            let totalButtonsWidth = buttons.reduce(0, { $0 + $1.frame.width }) + horizontalGap * CGFloat(buttons.count - 1)
+            if totalButtonsWidth < subviewWidth {
+                return layoutButtonsHorizontally()
+            }
         }
+        return layoutButtonsVertically()
+
     }
 
-    private func layoutAsActionSheet() {
-        header.frame.origin = CGPoint(x: horizontalPadding, y: verticalPadding)
+    private func layoutButtonsVertically() {
+        buttons.enumerated().forEach({ (arg) in
+            let (index, button) = arg
+
+            let fullButtonWidth = subviewWidth + horizontalPadding * 2
+
+            switch style {
+            case .actionSheet:
+                button.titleLabelOriginX = horizontalPadding
+            case .alert:
+                button.titleLabelOriginX = fullButtonWidth - horizontalPadding - button.bounds.width
+            }
+
+            button.frame.size.width = fullButtonWidth
+            button.frame.size.height += verticalPadding
+
+            if button == buttons.first {
+                button.frame.minY = CGFloat(text?.frame.maxY ?? header.frame.maxY) + verticalPadding
+            } else {
+                let previousElement = buttons[index - 1]
+                button.frame.origin.y = previousElement.frame.maxY
+            }
+
+        })
+    }
+
+    private func layoutButtonsHorizontally() {
         buttons.enumerated().forEach({ (arg) in
             let (index, button) = arg
-            let previousElement = (index > 0) ? buttons[index - 1] : header
             button.sizeToFit()
-            button.contentHorizontalAlignment = .left
-            button.bounds.size.height += verticalPadding
-            button.frame.size.width = max(bounds.size.width, button.frame.size.width)
-            button.frame.origin = CGPoint(x: 0, y: previousElement.frame.maxY)
+            button.clipsToBounds = true
```

Comment:
Just wondering if we should add the `button.clipsToBounds = true` for all buttons in vertical layout as well.
Also if it makes sense to add it all the subviews